SQL/proc_descriptions/procedure_compute_speeds_for_segments.md

```diff
@@ -0,0 +1,203 @@
+- the __name of the procedure__ + __arguments of the procedure__ implies that this procedure computes speeds for given hour + day_of_week in target area.
+- Input params of the procedure: `target_area_id::smallint`, `speeds_records_dataset`, `hour::smallint`, `day_of_week::smallint`. Returns nothing.
+- Flow:
+    - Create TEMP table `target_ways`, which is basically the same as return value of the function `get_ways_in_target_area(target_area_id smallint)`. But here we also add __INDEX__ to optimise search of the records. 
+    - Create TEMP table `node_sequences`. Table (`from_nodes_ways.node_id`, `to_node_ways.node_id`, `target_ways.id`, `from_nodes_ways.position`, `to_nodes_ways.position`). Basically creating records, which say from which __node__ to which __node__ you can get by which __way__ with additional info on __nodes' positions__ in a particular edge. Also some __INDEXes__ are added. 
+    - Main flow:
+        - __if-else statement__. If variable __day_of_week__ was not provided => __dataset_quality__=2, otherwise __dataset_quality__=1. This block creates TEMP table `grouped_speed_records` with (`from_osm_id`, `to_osm_id`, `speed`, `st_dev`) columns. If __dataset_quality__=1, then table `speed_records` is used as a source for TEMP table, otherwise table `speed_records_quarterly` is used.
+        - Save overall count of records from `nodes_ways_speeds` table to var __current_count__.
+        - __Insertion into nodes_ways_speeds__. Usage of union lets us to divide selection of data into 2 groups: ascending sequences, descending sequences, which are then merged by union.
+
+
+
+
+## TODO list
+- [x] Create description of the procedure
+- [x] Check if index `target_ways_id_idx` of `target_ways` is used. Is used
+- [x] Check if indexes `node_segments_osm_id_idx`, `node_segments_wf_idx`, `node_segments_wt_idx` of `node_sequences` table are used. Only `node_segments_wf_idx` is used -> Issue.
+- [x] Add deeper exaplanation of data selection for `grouped_speed_records` TEMP table. - No need, selection is plain.
+- [x] Replace `..., ...` in explanation of flow with classification name.
+- [ ] Reduction of TEMP tables into CTEs AKA With statements. - `target_ways`, `node_sequences` are candidates, not sure about `grouped_speed_records`.
+- [ ] Reduction of If-else statement with conditional WHERE clause with conditional WHERE clause. - Not easily achived as we have two different source for every case of dataset_quality, and refactored block with great chance would have worse perfomance. Does not worth it, in my opinion.
+- [ ] Reduction of calculation of average speed in 3rd step of main flow - it is calculated twice. May influence the execution perfomance depending on the number of stored data. - Not really sure, cause window function avg() applies after finding records, not before.
+
+## QA
+- Indexes `node_segments_osm_id_idx` & `node_segments_wt_idx` are not used, I think we should remove their creation queries.
```

Comment:
As I mentioned, these may be from some all-in import. In that case, we can keep it for now.